compiler/llvm/llvmAggregateGlobalOps.cpp

```diff
@@ -682,16 +683,15 @@ Instruction *AggregateGlobalOpsOpt::tryAggregating(Instruction *StartInst, Value
     // Determine alignment
     unsigned Alignment = Range.Alignment;
     if (Alignment == 0) {
-#if HAVE_LLVM_VER >= 150
-      Type *EltType = First->getType();
-#elif HAVE_LLVM_VER >= 130
-      Type *EltType =
-        cast<PointerType>(StartPtr->getType())->getPointerElementType();
-#else
-      Type *EltType =
-        cast<PointerType>(StartPtr->getType())->getElementType();
-#endif
-      Alignment = DL->getABITypeAlignment(EltType);
+      Type* eltType = nullptr;
+      if (LoadInst* ld = dyn_cast<LoadInst>(First)) {
+        eltType = ld->getType();
+      } else if (StoreInst* st = dyn_cast<StoreInst>(First)) {
+        eltType = st->getValueOperand()->getType();
+      }
+      if (eltType) {
```

Comment:
Seems like we might as well assert `eltType` is non-null, as previously we were functionally doing so by `cast`ing.